api/src/main/java/fr/catcore/server/translations/api/resource/language/DelegatedLanguage.java

```diff
@@ -0,0 +1,51 @@
+package fr.catcore.server.translations.api.resource.language;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.minecraft.client.font.TextVisitFactory;
+import net.minecraft.text.OrderedText;
+import net.minecraft.text.StringVisitable;
+import net.minecraft.text.Style;
+import net.minecraft.util.Language;
+
+import java.util.Optional;
+
+public final class DelegatedLanguage extends Language {
+    public static final DelegatedLanguage INSTANCE = new DelegatedLanguage();
+
+    private Language vanilla = Language.getInstance();
+
+    private DelegatedLanguage() {
+    }
+
+    public void setVanilla(Language language) {
+        this.vanilla = language;
+    }
+
+    @Override
+    public String get(String key) {
+        String override = ServerLanguageManager.INSTANCE.getSystemLanguage().get(key);
+        if (override != key) {
+            return override;
+        }
+        return this.vanilla.get(key);
```

Comment:
This should fix the last issue on server-side while not breaking client-side (Not tested) 
```suggestion
        return FabricLoader.getInstance().getEnvironment() == Environment.SERVER ? key : this.vanilla.get(key);
```